data_structures/src/serializers.rs

```diff
@@ -0,0 +1,461 @@
+extern crate flatbuffers;
+
+use crate::flatbuffers::protocol_generated::protocol::{
+    get_root_as_message, Address as FlatBufferAddress, AddressArgs, Command as FlatBuffersComand,
+    GetPeers as FlatBuffersGetPeers, GetPeersArgs, IpAddress as FlatBuffersIpAddress,
+    Ipv4 as FlatBuffersIpv4, Ipv4Args as FlatBuffersIpv4Args, Ipv6 as FlatBuffersIpv6,
+    Ipv6Args as FlatBuffersIpv6Args, Message as FlatBuffersMessage, MessageArgs,
+    Peers as FlatBuffersPeers, Peers, PeersArgs, Ping as FlatBuffersPing, PingArgs,
+    Pong as FlatBuffersPong, PongArgs, Verack as FlatBuffersVerack, VerackArgs,
+    Version as FlatBuffersVersion, VersionArgs,
+};
+use crate::types::{
+    Address, Command,
+    IpAddress::{Ipv4 as WitnetIpv4, Ipv6 as WitnetIpv6},
+    Message,
+};
+
+use flatbuffers::FlatBufferBuilder;
+
+const FTB_SIZE: usize = 1024;
+
+pub trait FromBytes<T> {
+    fn from_bytes(bytes: Vec<u8>) -> T;
+}
+
+pub trait ToBytes {
+    fn to_bytes(&self) -> Vec<u8>;
+}
+
+// #[cfg(feature = "flatbuffers")]
+impl FromBytes<Message> for Message {
+    fn from_bytes(bytes: Vec<u8>) -> Self {
+        // Get Flatbuffers Message
+        let message = get_root_as_message(&bytes);
+        let magic = message.magic();
+
+        // Create witnet's message to decode a flatbuffer message
+        match message.command_type() {
+            FlatBuffersComand::Ping => {
+                create_ping_message(message.command_as_ping().unwrap().nonce(), magic)
+            }
+            FlatBuffersComand::Pong => {
+                create_pong_message(message.command_as_pong().unwrap().nonce(), magic)
+            }
+            FlatBuffersComand::GetPeers => create_get_peers_message(magic),
+            FlatBuffersComand::Peers => {
+                create_peers_message(magic, message.command_as_peers().unwrap())
+            }
+            FlatBuffersComand::Verack => create_verack_message(magic),
+            FlatBuffersComand::Version => {
+                let command = message.command_as_version().unwrap();
+                let ftb_sender_address = command.sender_address().unwrap();
+                let ftb_receiver_address = command.receiver_address().unwrap();
```

Comment:
to be honest, I'm a bit scared with those `.unwrap()`.

Reply:
I have refactored almost all `.unwrap()`. I am working on refactor others.